### ISSUE TITLE
Clearing containers doesn't interoperate with PoolableDrawables

### ISSUE THREAD

**Issue by Flutterish** [ppy/osu-framework#4038]
Currently when you `Clear` a `Container`, it will `Dispose` of all of its contents, unless called with `false`.
This is a problem when the container contains both regular `Drawable`s and `PoolableDrawable`s as either the latter get `Disposed` or the former don't, which in turn requires us to write a filter function that `Remove`s and then `Return`s `PoolableDrawable`s before `Clear`ing the container.

I propose that the `Container<T>.Clear` method should do that on its own since `PoolableDrawable`s are a part of o!f.

This change could affect existing code in exactly one way I can think of: if a `PoolableDrawable` was stored in a variable, and a `Container` that contained it was called with `Clear(false)` and then that `PoolableDrawable` was added somewhere else - the difference would be that the `PoolableDrawable` was previously still considered as not freed while after the change it would.

**Comment by smoogipoo** [comment]
Yeah, this is a really big pain point. I'd say the best way is to expose a drawable property along the lines of `DisposeOnParentClear` rather than tying `CompositeDrawable` and `PoolableDrawable` together.

**Comment by peppy** [comment]
Should probably be `internal` for as long as we can manage it, too.

**Comment by Flutterish** [comment]
Perhaps a `protected Action` property that gets invoked on removal?
